app/src/services/ShopService.ts

```diff
@@ -440,7 +461,15 @@ export const ShopService: ShopControllerSocket & DashboardControllerSocket = {
       console.error('Invalid date, earlier than today')
       throw new InvalidParamsError()
     }
-    return ReservationRepository.insertReservation(reservationDate, clientId, shopId, menuId, stylistId)
+
+    const reservationDatesAndDuration = await ReservationRepository.fetchReservationsByDSS(shopId, stylistId)
+    const menuDuration = await ReservationRepository.fetchMenuDuration(menuId)
+    const availability = checkAvailablity(reservationDatesAndDuration, reservationDate, menuDuration).pop()
+
+    if (availability) {
+      return ReservationRepository.insertReservation(reservationDate, clientId, shopId, menuId, stylistId)
+    }
+    throw new UnavailableError()
```

Comment:
throw the error if it's not available, return the reservation when it's possible